.circleci/config.yml

```diff
@@ -1,56 +1,378 @@
+      - image: circleci/mongo:3.4.18
+  mssql:
+    docker:
+      - image: circleci/node:10.15.0
+      - image: "mcr.microsoft.com/mssql/server:2017-GA-ubuntu"
+        environment:
+          SA_PASSWORD: "Admin12345"
+          ACCEPT_EULA: "Y"
+  mysql:
+    docker:
+      - image: circleci/node:10.15.0
+      - image: mysql:5.7.24
         environment:
           MYSQL_ROOT_PASSWORD: "admin"
+          MYSQL_USER: "test"
+          MYSQL_PASSWORD: "test"
           MYSQL_DATABASE: "test"
+  oracle:
+    docker:
+      - image: circleci/node:10.15.0
+      # TODO oracle is skipped right now - not sure what image to use for the DB
+  postgres:
+    docker:
+      - image: circleci/node:10.15.0
       - image: circleci/postgres:9.6.11-postgis
         environment:
           POSTGRES_USER: "test"
           POSTGRES_PASSWORD: "test"
           POSTGRES_DB: "test"
-      - image: circleci/mongo:3.4.18
-      # - image: mcr.microsoft.com/mssql/server:2017-GA-ubuntu
-      #   environment:
-      #     SA_PASSWORD: "Admin12345"
-      #     ACCEPT_EULA: "Y"
-      # - image: cockroachdb/cockroach-unstable:latest
-      #   command: start --insecure
+  sqlite:
+    docker:
+      - image: circleci/node:10.15.0 # sqlite doesn't require another image
+  sqljs:
+    docker:
+      - image: circleci/node:10.15.0 # sqljs doesn't require another image
 
+commands:
+  prepare-for-tests:
+    parameters:
+      node_folder:
+        type: enum
+        enum:
+          - node8
+          - node10
+          - node12
     steps:
-      - checkout
+      - attach_workspace:
+          at: ~/typeorm/
       - run: sudo npm install -g npm@latest
       - run: cp ormconfig.circleci.json ormconfig.json
 
       # Download and cache dependencies
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "package.json" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+            - << parameters.node_folder >>-v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - << parameters.node_folder >>-v1-dependencies-
 
       - run: npm install
 
       - save_cache:
           paths:
             - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
-
-      # - run: docker-compose up -d
+          key: << parameters.node_folder >>-v1-dependencies-{{ checksum "package.json" }}
 
       # run tests
       - run: npm run lint
+
+  run-db-tests:
+    parameters:
+      database:
+        type: enum
+        enum:
+          - mariadb
+          - mongodb
+          - mssql
+          - mysql
+          - oracle
+          - postgres
+          - sqlite
+          - sqljs
+    steps:
+      - attach_workspace:
+          at: ~/typeorm/
+      - run: cp ormconfigs/ormconfig.<< parameters.database >>.json ormconfig.json
       - run: npm test
+
+jobs:
+  checkout_code:
```

Comment:
This is the first step that runs, which checks out the code, then copies it to separate folders for each node version. that way we can run `npm install` and `npm test` in the same node version-specific folder.